src/components/ProjectParticipants/ProjectParticipants.tsx

```diff
@@ -0,0 +1,92 @@
+import { FC, useCallback, useMemo, useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Text } from '@taskany/bricks';
+import { gapS, gapM, gray9, gray3 } from '@taskany/colors';
+
+import { trpc } from '../../utils/trpcClient';
+import { ActivityByIdReturnType, ProjectByIdReturnType } from '../../../trpc/inferredTypes';
+import { useProjectResource } from '../../hooks/useProjectResource';
+import { ModalEvent, dispatchModalEvent } from '../../utils/dispatchModal';
+import { UserEditableList } from '../UserEditableList/UserEditableList';
+import { SettingsCard } from '../SettingsContent';
+
+import { tr } from './ProjectParticipants.i18n';
+
+type User = NonNullable<ActivityByIdReturnType>;
+
+const StyledTitle = styled(Text)`
+    padding: ${gapS} ${gapM} ${gapM};
+`;
+
+const StyledListContianer = styled.div`
+    background-color: ${gray3};
+    padding: ${gapS} ${gapM};
+`;
+
+export const ProjectParticipants: FC<{
+    project: NonNullable<ProjectByIdReturnType>;
+}> = ({ project }) => {
+    const [userIdToRemove, setUserIdToRemove] = useState<string>('');
+    const filterIds = useMemo(() => project.participants.map(({ id }) => id) ?? [], [project]);
+
+    const { updateProject } = useProjectResource(project.id);
+
+    const { data: userToRemoveGoals, status } = trpc.project.getActivityGoals.useQuery(
+        {
+            id: project.id,
+            ownerId: userIdToRemove,
+        },
+        {
+            enabled: Boolean(userIdToRemove),
+        },
+    );
+
+    const onAdd = useCallback(
+        (user: User) => {
+            updateProject()({
+                ...project,
+                participants: [...project.participants, user],
+            });
+        },
+        [updateProject, project],
+    );
+
+    useEffect(() => {
+        if (userToRemoveGoals && userToRemoveGoals.length) {
```

Comment:
you can remove existence check `userToRemoveGoals` if set default value `const { data: userToRemoveGoals = [] }`